spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/core/converter/AbstractAzureMessageConverter.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.spring.integration.core.converter;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * Abstract class handles common conversion logic between <T> and {@link Message}
+ *
+ * @author Warren Zhu
+ */
+public abstract class AbstractAzureMessageConverter<T> implements AzureMessageConverter<T> {
+    private static ObjectMapper objectMapper = new ObjectMapper();
+
+    @Override
+    public T fromMessage(Message<?> message, Class<T> targetClass) {
```

Comment:
Would message conversion failure be possible? Shall we have a MessageConversionException anyway?

Reply:
We have `ConversionException` here.